src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/LSPProgressListener.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    internal abstract class LSPProgressListener
+    {
+        internal abstract bool TryListenForProgress(
+            string requestId,
+            Func<JToken, Task> onProgressResult,
+            TimeSpan timeoutAfterLastNotify,
```

Comment:
Do you envision this timeout changing in the future? Wondering if it makes sense to just make this an impl detail of the listener instead of requiring it to be passed in.

Reply:
I envision various handlers having different timeouts (ex. Roslyn returns completions every 100ms instead of 500ms for FAR). We'd want to accordingly react with a shorter `timeoutAfterLastNotify`.

Reply:
Ah really fair point then. Thanks for clarifying! To make it clear in the rest of this signature then I'd suggest changing this parameter to be `timeoutAfterLastResult` or the `onProgressResult` to be `onProgressNotify`